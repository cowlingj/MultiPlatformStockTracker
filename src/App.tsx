/** @format */

import StockList from "./components/StockList/View"
import * as React from "react"
import DispatcherFactory from "./components/StockList/Dispatcher/DispatcherFactory"
import Observable from "./archetecture/observer/Observable"
import {
  Increment,
  Decrement,
  RemoveItem,
  Init,
} from "./components/StockList/Messages"
import {
  quantChanged,
  DataModel,
  itemAdded,
  itemDeleted,
  initView,
} from "./components/StockList/Store"
import { DispatcherFactory as AdderDispatcherFactory } from "./components/Adder/Dispatcher/DispatcherFactory"
import { AddItem, Update, Init as AdderInit } from "./components/Adder/Messages"
import {
  itemAdded as adderItemAdded,
  update,
  initView as adderInitView,
} from "./components/Adder/Store"

const model: DataModel = { items: [] }

export default () => (
  <StockList
    {...{
      dispatcherFactory: new DispatcherFactory(
        new Observable<Increment>(),
        new Observable<Decrement>(),
        new Observable<RemoveItem>(),
        new Observable<Init>()
      ),
      inc: quantChanged(model),
      dec: quantChanged(model),
      add: itemAdded(model),
      del: itemDeleted(model),
      init: initView(model),
      adder: {
        dispatcherFactory: new AdderDispatcherFactory(
          new Observable<AddItem>(),
          new Observable<Update>(),
          new Observable<AdderInit>()
        ),
        add: adderItemAdded(),
        update: update(),
        init: adderInitView(),
      },
    }}
  />
)
