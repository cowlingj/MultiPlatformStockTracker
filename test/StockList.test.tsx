/** @format */

import renderer from "react-test-renderer"
import React from "react"
import App from "../src/App";

test("app matches snapshot", () => {
  expect(renderer.create(<App />)).toMatchSnapshot()
})
