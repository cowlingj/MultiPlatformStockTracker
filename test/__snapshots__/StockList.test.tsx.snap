// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app matches snapshot 1`] = `
<div
  className="rn-alignItems-1oszu61 rn-borderTopStyle-1efd50x rn-borderRightStyle-14skgim rn-borderBottomStyle-rull8r rn-borderLeftStyle-mm0ijv rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-display-6koalj rn-flexShrink-1qe8dj5 rn-flexBasis-1mlwlqe rn-flexDirection-eqz5dr rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-minHeight-ifefl9 rn-minWidth-bcqeeo rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-position-bnwqim rn-zIndex-1lgpqti"
>
  <div
    className="rn-alignItems-1oszu61 rn-borderTopStyle-1efd50x rn-borderRightStyle-14skgim rn-borderBottomStyle-rull8r rn-borderLeftStyle-mm0ijv rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-display-6koalj rn-flexShrink-1qe8dj5 rn-flexBasis-1mlwlqe rn-flexDirection-eqz5dr rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-minHeight-ifefl9 rn-minWidth-bcqeeo rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-position-bnwqim rn-zIndex-1lgpqti"
  >
    <input
      autoCapitalize="sentences"
      autoComplete="on"
      autoCorrect="on"
      className="rn-MozAppearance-97bpaa rn-WebkitAppearance-n1uzsy rn-backgroundColor-1niwhzg rn-borderTopColor-nqhrom rn-borderRightColor-1878er4 rn-borderBottomColor-mbada rn-borderLeftColor-fpul1g rn-borderTopLeftRadius-ou6ah9 rn-borderTopRightRadius-t12b5v rn-borderBottomRightRadius-zmljjp rn-borderBottomLeftRadius-pm2fo rn-borderTopStyle-1efd50x rn-borderRightStyle-14skgim rn-borderBottomStyle-rull8r rn-borderLeftStyle-mm0ijv rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-fontFamily-14xgk7a rn-fontSize-1b43r93 rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-resize-1dz5y72"
      data-focusable={true}
      dir="auto"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyPress={[Function]}
      onSelect={[Function]}
      readOnly={false}
      spellCheck={true}
      type="text"
      value=""
    />
    <div
      className="rn-alignItems-1oszu61 rn-backgroundColor-14sbq61 rn-borderTopLeftRadius-1iymjk7 rn-borderTopRightRadius-s2skl2 rn-borderBottomRightRadius-l5bh9y rn-borderBottomLeftRadius-101sy47 rn-borderTopStyle-1efd50x rn-borderRightStyle-14skgim rn-borderBottomStyle-rull8r rn-borderLeftStyle-mm0ijv rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-cursor-1loqt21 rn-display-6koalj rn-flexShrink-1qe8dj5 rn-flexBasis-1mlwlqe rn-flexDirection-eqz5dr rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-minHeight-ifefl9 rn-minWidth-bcqeeo rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-position-bnwqim rn-touchAction-1otgn73 rn-transitionDuration-eafdt9 rn-transitionProperty-1i6wzkk rn-userSelect-lrvibr rn-zIndex-1lgpqti"
      data-focusable={true}
      onKeyDown={[Function]}
      onKeyPress={[Function]}
      onKeyUp={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      role="button"
      tabIndex="0"
    >
      <div
        className="rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-color-jwli3a rn-display-1471scf rn-fontFamily-14xgk7a rn-fontSize-1b43r93 rn-fontStyle-o11vmf rn-fontVariant-ebii48 rn-fontWeight-majxgm rn-lineHeight-t9a87b rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-paddingTop-tskmnb rn-paddingRight-1pyaxff rn-paddingBottom-xd6kpl rn-paddingLeft-1m04atk rn-textAlign-q4m81j rn-textDecoration-bauka4 rn-textTransform-tsynxw rn-whiteSpace-q42fyq rn-wordWrap-qvutc0"
        dir="auto"
      >
        ADD
      </div>
    </div>
  </div>
</div>
`;
